dotCMS/src/main/java/com/dotcms/rest/IntegrityResource.java

```diff
@@ -230,79 +257,139 @@ public Response generateIntegrityData(@Context HttpServletRequest request, @Form
                 return Response.status(HttpStatus.SC_UNAUTHORIZED).build();
             }
 
-            ProcessStatus integrityDataGeneratorStatus = (ProcessStatus) servletContext.getAttribute("integrityDataGenerationStatus");
-
-            if(UtilMethods.isSet( integrityDataGeneratorStatus )) {
-                switch (integrityDataGeneratorStatus) {
-                    case PROCESSING:
-                        return Response.status(HttpStatus.SC_PROCESSING).build();
-                    case FINISHED:
-                        StreamingOutput output = new StreamingOutput() {
-                            public void write(OutputStream output) throws IOException, WebApplicationException {
-                                InputStream is = Files.newInputStream(Paths.get(
-                                        ConfigUtils.getIntegrityPath() + File.separator
-                                                + requesterEndPoint.getId() + File.separator
-                                                + INTEGRITY_DATA_TO_CHECK_ZIP_FILE_NAME));
-
-                                byte[] buffer = new byte[1024];
-                                int bytesRead;
-                                //read from is to buffer
-                                while((bytesRead = is.read(buffer)) !=-1){
-                                    output.write(buffer, 0, bytesRead);
-                                }
-                                is.close();
-                                //flush OutputStream to write any buffered data to file
-                                output.flush();
-                                output.close();
-
-                            }
-                        };
-                        return Response.ok(output).build();
+            if (integrityMetadata.get().getProcessStatus() == ProcessStatus.PROCESSING) {
+                Logger.info(
+                        IntegrityResource.class,
+                        String.format(
+                                "Receiver at %s:> integrity data generation for endpoint id %s still ongoing therefore it's not ready and need to wait",
+                                localAddress,
+                                requesterEndpoint.getId()));
+                return Response.status(HttpStatus.SC_PROCESSING).build();
+            } else if (integrityMetadata.get().getProcessStatus() == ProcessStatus.FINISHED &&
+                    IntegrityUtil.doesIntegrityDataFileExist(
+                            requesterEndpoint.getId(),
+                            IntegrityUtil.INTEGRITY_DATA_TO_CHECK_ZIP_FILENAME)) {
+                final String zipFilePath = IntegrityUtil.getIntegrityDataFile(
+                        requesterEndpoint.getId(),
+                        IntegrityUtil.INTEGRITY_DATA_TO_CHECK_ZIP_FILENAME);
+                final StreamingOutput output = so -> {
+                    final InputStream is = Files.newInputStream(Paths.get(zipFilePath));
```

Comment:
Codacy found an issue: [Avoid variables with short names like is](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6151243)